libbeat/template/template.go

```diff
@@ -41,6 +45,39 @@ func New(beatVersion string, beatName string, esVersion string, config TemplateC
 		pattern = name + "-*"
 	}
 
+	compiler := func(key string, ops []fmtstr.VariableOp) (fmtstr.FormatEvaler, error) {
+		if key == "[beat.name]" {
+			return Formatter{s: beatName}, nil
+		}
+
+		if key == "[beat.version]" {
+			return Formatter{s: bV.String()}, nil
+		}
```

Comment:
Date patterns start with `+` character. See: https://github.com/elastic/beats/blob/master/libbeat/common/fmtstr/formatevents.go#L252

That is, we can replace `%{+<pattern>}` with `*` in here.